From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sadaf Ebrahimi <sadafebrahimi@google.com>
Date: Mon, 23 May 2022 22:34:43 +0000
Subject: [PATCH] Prevent integer overflow in copyString

Bug: http://b/221384482
Change-Id: Ibdcb5dc24ee8886a04c2e29bd6ddccf29ece73ad
(cherry picked from commit e25c84037506951dfe74a5fae1627fe22bc0ebf4)
Merged-In: Ibdcb5dc24ee8886a04c2e29bd6ddccf29ece73ad
---
 lib/xmlparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/xmlparse.c b/lib/xmlparse.c
index 90a237f3..67f661b5 100644
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -7175,7 +7175,7 @@ static XML_Char *
 copyString(const XML_Char *s,
            const XML_Memory_Handling_Suite *memsuite)
 {
-    int charsRequired = 0;
+    size_t charsRequired = 0;
     XML_Char *result;
 
     /* First determine how long the string is */
